ledger-api/grpc-definitions/com/daml/ledger/api/v1/completion.proto

```diff
@@ -32,6 +32,37 @@ message Completion {
   // Optional
   string transaction_id = 3;
 
+  // The application ID that was used for the submission, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  // TODO(v2): Can be made required once all Daml ledgers return the application ID
+  string application_id = 6;
+
+  // The set of parties on whose behalf the commands were executed.
+  // Contains the union of `party` and `act_as` from ``commands.proto``.
+  // The order of the parties need not be the same as in the submission.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional
+  // TODO(v2): Can be made required once all Daml ledgers return the application ID
+  repeated string act_as = 7;
+
+  // The submission ID this completion refers to, as described in ``commands.proto``.
+  // Optional
+  // TODO(v2): Can be made required once all Daml ledgers return the application ID
+  // TODO(v2): Do we want to enforce the format of a UUID here?
```

Comment:
What do we gain by enforcing a UUID format?

I'd suggest we only limit the length, and otherwise treat it as an opaque string.

Reply:
No operational benefits. It just conveys better the semantic meaning that this should be universally unique.

Reply:
Let's leave it as a ledger string.